src/net.c

```diff
@@ -1044,80 +1044,67 @@ int
 coap_client_delay_first(coap_session_t *session)
 {
   if (session->type == COAP_SESSION_TYPE_CLIENT && session->doing_first) {
-#ifndef WITH_LWIP
     if (session->doing_first) {
-      unsigned ref;
       int timeout_ms = 5000;
+ static int recursive = 0;
```

Comment:
I wonder if `recursive` being static calls for issues when multiple sessions are handled concurrently. Maybe this should be local to the current session?

Reply:
Good point.  However, we currently only support single threading per context, so this would be a multiple context issue.  Will fix.